services/analytics/src/main/scala/com/wedriveu/services/analytics/entity/VehiclesCounterAlgorithm.scala

```diff
@@ -0,0 +1,35 @@
+package com.wedriveu.services.analytics.entity
+import com.wedriveu.services.shared.entity.VehicleCounter
+import collection.mutable._
+import collection.JavaConverters._
+
+
+/**
+  * @author Stefano Bernagozzi
+  */
+
+//trait VehiclesCounterAlgorithm {
+//  def vehicleCounter(vehicleList: java.util.List[AnalyticsVehicle]): VehicleCounter
+//}
+ class VehiclesCounterAlgorithm {
+  //object VehiclesCounterAlgorithmImpl { //extends VehiclesCounterAlgorithm {
+    def vehicleCounter(vehicleList: java.util.List[AnalyticsVehicle]): VehicleCounter = {
```

Comment:
I don't know if this is the case, but maybe you could do a tail recursive function, see:

```
override def zipRight: List[(A,Int)] = {
    @annotation.tailrec
    def innerZipRight(origin: List[A], dest: List[(A,Int)], index: Int): List[(A,Int)] = origin match {
      case h :: t => innerZipRight(t, dest append List((h, index)), index + 1)
      case _ => dest
    }
    innerZipRight(this, List(), 0)
  }
```